homeassistant/components/growatt_server/sensor.py

```diff
@@ -46,8 +47,7 @@ def get_device_list(api, config):
         not login_response["success"]
         and login_response["msg"] == LOGIN_INVALID_AUTH_CODE
     ):
-        _LOGGER.error("Username, Password or URL may be incorrect!")
-        return
+        raise ConfigEntryError("Username, Password or URL may be incorrect!")
```

Comment:
`ConfigEntryError` should only be raised during config entry setup in the init module or from a coordinator (for the first refresh), to indicate that we can't setup the config entry. This exception should not be used for other purposes or at other times.

The integration should probably be refactored to move most of the setupaand initial data fetching to the init module `async_setup_entry` and just create the entities here in the sensor platform.

For now raise `HomeAssistantError` with a message as argument and catch that when calling `get_device_list` and log the error message and return.